charts/alertmanager/templates/_helpers.tpl

```diff
@@ -53,24 +53,26 @@ Return  the proper Storage Class
 {{/*
 Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
 but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
+This refactoring prioritizes persistence.storageClassName over global.storageClass.
 */}}
-{{- if .Values.global.storageClass -}}
-    {{- if (eq "-" .Values.global.storageClass) -}}
+{{- if .Values.persistence.storageClassName -}}
+    {{- if (eq "~" .Values.persistence.storageClassName) -}}
```

Comment:
In YAML, `~` is `null` <https://yaml.org/spec/1.2-old/spec.html#id2805071> so I think we can get away with `if not .Values.persistence.storageClassName`

Reply:
done